frontend/coprs_frontend/coprs/views/apiv3_ns/schema.py

```diff
@@ -0,0 +1,560 @@
+    "isolation": isolation_field,
+}
+
+project_chroot_build_config_model = \
+    api.model("ProjectChrootBuildConfig", project_chroot_build_config_schema)
+
+source_dict_scm_schema = {
+    "clone_url": clone_url_field,
+    "committish": committish_field,
+    "source_build_method": source_build_method_field,
+    "spec": spec_field,
+    "subdirectory": subdirectory_field,
+    "type": scm_type_field,
+}
+
+source_dict_scm_model = api.model("SourceDictSCM", source_dict_scm_schema)
+
+source_dict_pypi_schema = {
+    "pypi_package_name": pypi_package_name_field,
+    "pypi_package_version": pypi_package_version_field,
+    "spec_generator": pypi_spec_generator_field,
+    "spec_template": pypi_spec_template_field,
+    "python_versions": pypi_versions_field,
+}
+
+source_dict_pypi_model = api.model("SourceDictPyPI", source_dict_pypi_schema)
+
+source_package_schema = {
+    "name": packagename_field,
+    "url": source_package_url_field,
+    "version": source_package_version_field,
+}
+
+source_package_model = api.model("SourcePackage", source_package_schema)
+
+build_schema = {
+    "chroots": chroots_field,
+    "ended_on": ended_on_field,
+    "id": id_field,
+    "is_background": is_background_field,
+    "ownername": ownername_field,
+    "project_dirname": project_dirname_field,
+    "projectname": projectname_field,
+    "repo_url": repo_url_field,
+    "source_package": Nested(source_package_model),
+    "started_on": started_on_field,
+    "state": state_field,
+    "submitted_on": submitted_on_field,
+    "submitter": submitter_field,
+}
+
+build_model = api.model("Build", build_schema)
+
+package_builds_schema = {
+    "latest": Nested(build_model, allow_null=True),
+    "latest_succeeded": Nested(build_model, allow_null=True),
+}
+
+package_builds_model = api.model("PackageBuilds", package_builds_schema)
+
+# TODO We use this schema for both GetPackage and PackageEdit. The `builds`
+# field is returned for both but only in case of GetPackage it can contain
+# results. How should we document this?
+package_schema = {
+    "id": id_field,
+    "name": packagename_field,
+    "projectname": projectname_field,
+    "ownername": ownername_field,
+    "source_type": source_type_field,
+    # TODO Somehow a Polymorh should be used here for `source_dict_scm_model`,
+    # `source_dict_pypi_model`, etc. I don't know how, so leaving an
+    # undocumented value for the time being.
+    "source_dict": Raw,
+    "auto_rebuild": auto_rebuild_field,
+    "builds": Nested(package_builds_model),
+}
+
+package_model = api.model("Package", package_schema)
+
+
+def clone(field):
+    """
+    Return a copy of a field
+    """
+    kwargs = field.__dict__.copy()
+    return field.__class__(**kwargs)
+
+
+def optional(field):
+    """
+    Unless this is an error on my side, there is a bug in flask-restx displaying
+    all URL path parameters inside the Swagger documentation as as required.
+    This is a temporary hack until it gets fixed.
```

Comment:
Has this been reported?

Reply:
As it turned out, it was an error on my side. Updated.